CMakeLists.txt

```diff
@@ -451,7 +451,32 @@ endif ()
 install (DIRECTORY ${P4C_SOURCE_DIR}/p4include
   DESTINATION ${P4C_ARTIFACTS_OUTPUT_DIRECTORY})
 
+################################ Linters Begin ################################
+
+file(
+  GLOB_RECURSE P4C_LINT_LIST
+  backends/*.cpp
+  backends/*.h
+  control-plane/*.cpp
+  control-plane/*.h
+  frontends/*.cpp
+  frontends/*.h
+  extensions/*.cpp
+  extensions/*.h
+  ir/*.cpp
+  ir/*.h
+  tools/*.cpp
+  tools/*.h
+)
+# Filter some folders from the CI checks.
+list(FILTER P4C_LINT_LIST EXCLUDE REGEX "backends/p4tools/submodules")
```

Comment:
Is this how extensions should declare exclusions too?
Maybe you can add a comment if that's correct.

Reply:
This is done specifically for the compiler files. Extensions are free to add files how they see fit.